packages/zoe/src/contractFacet/types.js

```diff
@@ -59,6 +60,9 @@
  */
 
 /**
+ * @deprecated reallocate() will be supported until at least 2023/09/01. It may
+ * be removed without further warning any time after 2023/11/01.
```

Comment:
Wow, I love the explicitness! Where do these dates come from? Can we do likewise for other deprecations?

Reply:
I completely made them up. I think we should be this explicit. I'm open to discussion on plausible dates.

Reply:
we should remove - not just deprecate - as rapidly as possible. When MN-2 contracts go live the process to deprecate will definitely need additional communication and a schedule. Before MN-2 all the on-chain code potentially using `reallocate` is in this repo and easily found.

Reply:
> we should remove - not just deprecate - as rapidly as possible. When MN-2 contracts go live the process to deprecate will definitely need additional communication and a schedule. Before MN-2 all the on-chain code potentially using reallocate is in this repo and easily found.

I think we've gotten to the stage where we ought to at least consider the impact on code written by third parties. OTOH, this is a change that's relatively trivial for them to adapt to if, at some point, they discover just before deploying that they haven't migrated. It's a change from `atomicRearrange(zcf, ...)` to `zcf.atomicRearrange(...)`, so it's always a 1-line change that doesn't propagate.

#7901, #7903, and #7904 show that deployed contracts need an upgrade to make the change.

> What further warning do you have in mind?

My point was that if we were to push it faster (as @ivanlei may be advocating), we'd make announcements on dev channels. The intent was that this warning was sufficient notice that if we do it later, no further announcements should be expected. I'm not stuck on this formulation.

Reply:
I mis-spoke above. This deprecation notice is on `rearrange`, which is the make-it-so step in the old (stagings) approach that we've been trying to get rid of.  Any third-party code that has been around for a while is probably still written this way, and it's not quite trivial to upgrade to the new API.